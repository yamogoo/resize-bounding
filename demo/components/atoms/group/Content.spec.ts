import { describe, expect, test } from "vitest";
import { mount } from "@vue/test-utils";

import Content from "./Content.vue";

describe("Content", () => {
  test('should have "--h" postfix in class name by default', () => {
    const wrapper = mount(Content);

    const classes = wrapper
      .find('[data-testid="ui-content-group"]')
      .classes()
      .toString();
    expect(classes).toContain(`--h`);
    expect(classes).toMatchSnapshot();
  });

  test.each(["h", "v"])(
    "should contain %s postfix in class name",
    (direction) => {
      const wrapper = mount(Content, {
        props: {
          direction: direction as "h" | "v",
        },
      });

      const classes = wrapper
        .find('[data-testid="ui-content-group"]')
        .classes()
        .toString();
      expect(classes).toContain(`--${direction}`);
      expect(classes).toMatchSnapshot();
    },
  );

  test.each(['<p data-testid="default-content">Default Slot Content</p>'])(
    "should render default slot with content (%s)",
    (defaultSlot) => {
      const wrapper = mount(Content, {
        slots: {
          default: defaultSlot,
        },
      });

      const html = wrapper.find('[data-testid="default-content"]').html();

      expect(html).toContain(defaultSlot);
      expect(html).toMatchSnapshot();
    },
  );
});
