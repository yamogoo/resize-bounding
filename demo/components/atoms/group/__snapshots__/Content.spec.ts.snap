// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Content > should contain h postfix in class name 1`] = `"ui-content-group,--h,--start"`;

exports[`Content > should contain v postfix in class name 1`] = `"ui-content-group,--v,--start"`;

exports[`Content > should have "--h" postfix in class name by default 1`] = `"ui-content-group,--h,--start"`;

exports[`Content > should render default slot with content (<p data-testid="default-content">Default Slot Content</p>) 1`] = `"<p data-testid="default-content">Default Slot Content</p>"`;
